AzureMonitorAgent/agent.py

```diff
@@ -1190,6 +1190,7 @@ def find_package_manager(operation):
             break
 
     if PackageManager == "":
+        exit_if_vm_not_supported(operation)
```

Comment:
seems maybe better to alter `install()` and `uninstall()` so that we call `exit_if_vm_not_supported` before `find_package_manager` ?

Reply:
Sounds good - I modified the check to be before `find_package_manager` in both options